raft-tech/TANF-app
# Issue 968: Codecov Uploader update and validation

### Summary of Changes
Closes #968 

Currently we have no protections in place to assert that the downloaded Codecov uploader has not been tampered with prior to uploading coverage reports. Additionally, the Codecov BASH uploader that we currently use [is being deprecated](https://docs.codecov.com/docs/about-the-codecov-bash-uploader). This PR replaces usage of that uploader with the [new compiled uploader binary](https://docs.codecov.com/docs/codecov-uploader) and ensures that all integrity checks are ran prior to installation on the system.

### How to Test
* Re-run the Circle CI tests for this branch
* Confirm Codecov upload is completed in Circle CI job output
* You should also be able to see the upload shortly after in the [Codecov UI](https://app.codecov.io/gh/raft-tech/TANF-app/commits?page=1)

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

  - [x] Codecov bash uploader script is validated to ensure that it has not been tampered with before allowing it to execute in CI

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

N/A, CI/CD changes only

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*


### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

N/A, CI/CD changes only

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

> NOTE: until we have a proper staging environment this may not be satisfiable prior to merging

N/A, CI/CD changes only

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

N/A, CI/CD changes only

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*
